--- a/gui/wxmain.cc	2021-03-13 17:54:06.000000000 +0800
+++ b/gui/wxmain.cc	2022-09-06 08:33:33.000000000 +0800
@@ -93,9 +93,7 @@
 //#include "bitmaps/configbutton.xpm"
 #include "bitmaps/userbutton.xpm"
 #include "bitmaps/saverestore.xpm"
-#ifdef __WXGTK__
 #include "icon_bochs.xpm"
-#endif
 
 // FIXME: ugly global variable that the bx_gui_c object in wx.cc can use
 // to access the MyFrame.

--- a/configure	2021-08-01 16:12:28.000000000 +0800
+++ b/configure	2022-09-07 23:10:03.000000000 +0800
@@ -24972,25 +24972,16 @@
   # pkg-config is required to set TOOLKIT_CXXFLAGS and LIBS
   if test "$PKGCONFIG" != not_found; then
     # if wxGTK is based on GTK 3.0, use it for gui debugger to avoid conflicts
-    if test "$with_wx" = yes -a "$wx_needs_gdk_version" = 3; then
-      pkg-config --exists gtk+-3.0
-      if test x$? = x0; then
-        TOOLKIT_CXXFLAGS="`pkg-config --cflags gtk+-3.0`"
-        LIBS="$LIBS `pkg-config --libs gtk+-3.0`"
-        bx_have_gtk_version=3
-      fi
-    else
-      pkg-config --exists gtk+-2.0
-      if test x$? = x0; then
-        TOOLKIT_CXXFLAGS="`pkg-config --cflags gtk+-2.0`"
-        LIBS="$LIBS `pkg-config --libs gtk+-2.0`"
-        bx_have_gtk_version=2
-      fi
-    fi
+    TOOLKIT_CXXFLAGS="`pkg-config --cflags gtk+-3.0`"
+    LIBS="$LIBS `pkg-config --libs gtk+-3.0`"
+    bx_have_gtk_version=3
   else
     echo "ERROR: pkg-config was not found, unable to access the gtk+-2.0 package."
     echo "The gui debugger support is disabled now."
   fi
+else
+  { $as_echo "$as_me:${as_lineno-$LINENO}: need_gtk2 != 1" >&5
+$as_echo_n "need_gtk2 != 1" >&2; }
 fi
 
 if test "$display_libs" = " wxWidgets"; then
@@ -25231,6 +25222,14 @@
 ENH_DBG_OBJS=""
 if test "$gui_debugger" = 1; then
   if test "$bx_have_gtk_version" -ge 2; then
+      { $as_echo "$as_me:${as_lineno-$LINENO}: bx_have_gtk_version >= 2" >&5
+    $as_echo "bx_have_gtk_version >= 2" >&2; }
+  else
+      { $as_echo "$as_me:${as_lineno-$LINENO}: bx_have_gtk_version < 2" >&5
+    $as_echo "bx_have_gtk_version < 2" >&2; }
+  fi
+
+  if test "$bx_have_gtk_version" -ge 2; then
     ENH_DBG_OBJS="enh_dbg.o gtk_enh_dbg_osdep.o"
     $as_echo "#define BX_DEBUGGER_GUI 1" >>confdefs.h
 
@@ -25239,8 +25238,8 @@
     $as_echo "#define BX_DEBUGGER_GUI 1" >>confdefs.h
 
   else
-    { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: The Bochs debugger gui cannot be compiled here, disabling it" >&5
-$as_echo "$as_me: WARNING: The Bochs debugger gui cannot be compiled here, disabling it" >&2;}
+    { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: The Bochs debugger gui cannot be compiled here, disabling it 1" >&5
+$as_echo "$as_me: WARNING: The Bochs debugger gui cannot be compiled here, disabling it 1" >&2;}
     $as_echo "#define BX_DEBUGGER_GUI 0" >>confdefs.h
 
   fi
